modules/contracts/apps/challenge/src/handlers/execute.rs

```diff
@@ -152,349 +163,228 @@ fn update_friends_for_challenge(
 ) -> AppResult {
     app.admin.assert_admin(deps.as_ref(), &info.sender)?;
     let challenge = CHALLENGE_LIST.load(deps.storage, challenge_id)?;
-    if challenge.status != ChallengeStatus::Active {
-        return Err(AppError::WrongChallengeStatus {});
-    }
+    // Validate friend addr and account ids
+    let friends_validated: Vec<(Addr, Friend<Addr>)> = friends
+        .iter()
+        .cloned()
+        .map(|human| human.check(deps.as_ref(), app))
+        .collect::<AbstractSdkResult<_>>()?;
+
+    let (voters_addrs, friends): (Vec<Addr>, Vec<Friend<Addr>>) =
+        friends_validated.into_iter().unzip();
+
+    let proposal_id = challenge.current_proposal_id;
+
+    SIMPLE_VOTING
+        .load_proposal(deps.storage, proposal_id)?
+        .assert_ready_for_action(&env.block)?;
 
     match op_kind {
-        UpdateFriendsOpKind::Add => {
-            add_friends_for_challenge(deps, info, app, challenge_id, friends)
+        UpdateFriendsOpKind::Add {} => {
+            CHALLENGE_FRIENDS.update(deps.storage, challenge_id, |current_friends| {
+                let mut current_friends = current_friends.ok_or(AppError::ZeroFriends {})?;
+                ensure!(
+                    friends.len() + current_friends.len() < MAX_AMOUNT_OF_FRIENDS as usize,
+                    AppError::TooManyFriends {}
+                );
+                current_friends.extend(friends.clone().into_iter());
+                AppResult::Ok(current_friends)
+            })?;
+            SIMPLE_VOTING.add_voters(deps.storage, proposal_id, &env.block, &voters_addrs)?;
         }
-        UpdateFriendsOpKind::Remove => {
-            remove_friends_from_challenge(deps, info, app, challenge_id, friends)
+        UpdateFriendsOpKind::Remove {} => {
+            CHALLENGE_FRIENDS.update(deps.storage, challenge_id, |current_friends| {
+                let mut current_friends = current_friends.ok_or(AppError::ZeroFriends {})?;
+                for rem_friend in friends.iter() {
+                    current_friends.retain(|friend| friend != rem_friend);
+                }
+                AppResult::Ok(current_friends)
```

Comment:
Yeah, it's not possible to reach voting "threshold" in case of 0 friends, so it just will be counted as successful challenge.
But it's just a obviously bad usage of an app, so idk if we should care about it :)